app/models/task.py

```diff
@@ -1,5 +1,50 @@
 from app import db
 
-
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String, nullable=False)
+    description = db.Column(db.String, nullable=False)
+    completed_at = db.Column(db.DateTime, nullable=True)
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.goal_id'), nullable=True)
+    goal = db.relationship("Goal", back_populates="tasks")
+
+    def to_dict(self):
+        if self.completed_at:
+            return {
+                "id": self.task_id,
+                "title": self.title,
+                "description": self.description,
+                "is_complete": True,
+            }
+        else:
+            return {
+                "id": self.task_id,
+                "title": self.title,
+                "description": self.description,
+                "is_complete": False,
+            }
```

Comment:
Rather than create a conditional that returns a sliiiiightly different object depending on the value of self.completed_at, a better idea would be to define the is_complete attribute by the existence of completed_at. That might look similar to the following:

     return {
          "id": self.task_id,
          "title": self.title,
          "description": self.description,
          "is_complete": self.completed_at != None,
     }